mitodl/open-discussions
# Nl/article post backend

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of #1436 

#### What's this PR do?
Adds a new post type "article".
- Adds an `Article` model that stores a JSON blob representing the article content. This will be managed by the frontend so it just needs to be valid JSON
- Moves much of the `sync_*` logic into transactions within the api itself, so we have guarantees of the creation timing of these objects.

I split the changes between fixture changes and actual code changes. The code is in commit https://github.com/mitodl/open-discussions/pull/1452/commits/14794ff3a33110a1b31adef66e041f3aa7d3199f.

#### How should this be manually tested?
Go to `/api/v0/channels/abc/posts/` and craft a POST request that includes a "`article_content`" field that is valid JSON. Verify that this request succeeds and the JSON is return in the reponse and that other post APIs still work for old post types and this new one.
